numpy/core/include/numpy/simd/simd.h

```diff
@@ -0,0 +1,45 @@
+#ifndef _NPY_SIMD_H_
+#define _NPY_SIMD_H_
+/**
+ * the NumPy C SIMD vectorization interface "NPYV" are types and functions intended
+ * to simplify vectorization of code on different platforms, currently supports
+ * the following SIMD extensions SSE, AVX2, AVX512, VSX and NEON.
+ *
+ * TODO: Add an independent sphinx doc.
```

Comment:
+1. Perhaps a README.md in this directory describing the overall design could provide the basis for more documentation

Reply:
Soon or later we will need to support CPP, so would it better if we could use Doxygen instead of sphix or md to generate documentation from one single header e.g. "simd.hpp" that wrapping all the defined NPYV C intrinsics, which will support overloading, operators, type-safety, etc. what do you think? NumPy really needs to support a fully Doxygen format.

Reply:
If you prefer to write in Doxygen format, that may be fine. It will need to be integrated with Sphinx at some point, but we can decide later (imho that doesn't need to be in this PR). The [Exhale](https://exhale.readthedocs.io/en/latest/) or [Breathe](https://breathe.readthedocs.io/en/latest/) Sphinx extensions could be used to achieve this perhaps.

Due to the large amount of existing content (and the need to make it look good) we are unlikely to ever move away from Sphinx, so either integrating via such an extension or copying and changing formatting to reST are the options.